xrdp/xrdp_tconfig.c

```diff
@@ -199,6 +199,70 @@ tconfig_load_gfx_x264_ct(toml_table_t *tfile, const int connection_type,
     return 0;
 }
 
+static int tconfig_load_gfx_order(toml_table_t *tfile, struct xrdp_tconfig_gfx *config)
+{
+    /*
+     * This config loader is not responsible to check if xrdp is built with
+     * H264/RFX support. Just loads configurations as-is.
+     */
+
+    TCLOG(LOG_LEVEL_TRACE, "[codec]");
+
+    int h264_found = 0;
+    int rfx_found = 0;
+
+    config->codec.h264_idx = -1;
+    config->codec.rfx_idx = -1;
+
+    toml_table_t *codec = toml_table_in(tfile, "codec");
+    toml_array_t *order = toml_array_in(codec, "order");
+
+    if (codec && order)
+    {
+        for (int i = 0; ; i++)
+        {
+            toml_datum_t datum = toml_string_at(order, i);
+
+            if (datum.ok)
+            {
+                if (g_strcasecmp(datum.u.s, "h264") == 0 ||
+                        g_strcasecmp(datum.u.s, "h.264") == 0)
```

Comment:
This is a slightly oddball case, but you might want to consider checking `h264_found` is zero here. Otherwise this string might not work how you want:-

```
order = [ "RFX", "H264" , "RFX" ]
```

Reply:
You're right.